# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadHandler.h
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c dom/script/ScriptLoadHandler.h
--- a/dom/script/ScriptLoadHandler.h	Fri Feb 18 12:45:52 2022 +0000
+++ b/dom/script/ScriptLoadHandler.h	Fri Feb 18 12:45:53 2022 +0000
@@ -17,20 +17,23 @@
 #include "mozilla/RefPtr.h"
 #include "mozilla/UniquePtr.h"
 
+namespace JS::loader {
+class ScriptLoadRequest;
+}
+
 namespace mozilla {
 
 class Decoder;
 
 namespace dom {
 
-class ScriptLoadRequest;
 class ScriptLoader;
 class SRICheckDataVerifier;
 
 class ScriptLoadHandler final : public nsIIncrementalStreamLoaderObserver {
  public:
   explicit ScriptLoadHandler(
-      ScriptLoader* aScriptLoader, ScriptLoadRequest* aRequest,
+      ScriptLoader* aScriptLoader, JS::loader::ScriptLoadRequest* aRequest,
       UniquePtr<SRICheckDataVerifier>&& aSRIDataVerifier);
 
   NS_DECL_ISUPPORTS
@@ -96,7 +99,7 @@
   RefPtr<ScriptLoader> mScriptLoader;
 
   // The ScriptLoadRequest for this load. Decoded data are accumulated on it.
-  RefPtr<ScriptLoadRequest> mRequest;
+  RefPtr<JS::loader::ScriptLoadRequest> mRequest;
 
   // SRI data verifier.
   UniquePtr<SRICheckDataVerifier> mSRIDataVerifier;